packages/data/README.md

```diff
@@ -139,67 +139,125 @@ The `controls` option should be passed as an object where each key is the name o
+	let storeChanged = () => {};
+	const prices = { hammer: 7.50 };
 
-dispatch( 'my-shop' ).setPrice( 'hammer', 9.75 );
+	const selectors = {
+		getPrice( itemName ): {
+			return prices[ itemName ];
+		},
+	};
+
+	const actions = {
+		setPrice( itemName, price ): {
+			prices[ itemName ] = price;
+			storeChanged();
+		},
+	};
+
+	return {
+		getSelectors() {
+			return selectors;
+		},
+		getActions() {
+			return actions;
+		},
+		subscribe( listener ) {
+			storeChanged = listener;
+		}
+	};
+}
+
+registry.registerGenericStore( 'custom-data', createCustomStore() );
 ```
 
-#### `subscribe(): Function`
+## Comparison with Redux
 
-Given a listener function, the function will be called any time the state value of one of the registered stores has changed. This function returns a `unsubscribe` function used to stop the subscription.
+The data module shares many of the same [core principles](https://redux.js.org/introduction/three-principles) and [API method naming](https://redux.js.org/api-reference) of [Redux](https://redux.js.org/). In fact, it is implemented atop Redux. Where it differs is in establishing a modularization pattern for creating separate but interdependent stores, and in codifying conventions such as selector functions as the primary entry point for data access.
 
-_Example:_
+The [higher-order components](#higher-order-components) were created to complement this distinction. The intention with splitting `withSelect` and `withDispatch` — where in React Redux they are combined under `connect` as `mapStateToProps` and `mapDispatchToProps` arguments — is to more accurately reflect that dispatch is not dependent upon a subscription to state changes, and to allow for state-derived values to be used in `withDispatch` (via [higher-order component composition](/packages/compose/README.md)).
 
-```js
-const { subscribe } = wp.data;
+Specific implementation differences from Redux and React Redux:
 
-const unsubscribe = subscribe( () => {
-	// You could use this opportunity to test whether the derived result of a
-	// selector has subsequently changed as the result of a state update.
-} );
+-   In Redux, a `subscribe` listener is called on every dispatch, regardless of whether the value of state has changed.
+    -   In `@wordpress/data`, a subscriber is only called when state has changed.
+-   In React Redux, a `mapStateToProps` function must return an object.
+    -   In `@wordpress/data`, a `withSelect` mapping function can return `undefined` if it has no props to inject.
+-   In React Redux, the `mapDispatchToProps` argument can be defined as an object or a function.
+    -   In `@wordpress/data`, the `withDispatch` higher-order component creator must be passed a function.
 
-// Later, if necessary...
-unsubscribe();
-```
+## API
+
+It is very rare that you should access store methods directly. Instead, the following suite of functions and higher-order components is provided for the most common data access and manipulation needs.
```

Comment:
I'm struggling to remember the original context under which the phrasing was meant to apply: "It is very rare that you should access store methods directly" ([original document](https://github.com/WordPress/gutenberg/blob/f2bf8a1c66eb80cbf30c29c1cc00b08ad98bd9df/data/README.md#data-access-and-manipulation)). In any case, I think it reads even less appropriately now than ever, considering that the following set of API documentation are not really organized in any way, and are simply a listing of all exports from the module. I might suggest the whole paragraph be removed.

Reply:
As someone who has been a significant consumer of this api, I agree with @aduth's comment here.  Another thing I'm not sure about with regards to this documentation change is the order of things.  I think the part about generic stores should remain near the end of the documentation as it is a much more advanced api and thus would be referenced less often.

Reply:
Removed at 153a6b4